File: etamen-flutter/lib/features/radiology/domain/entities/radiology_result.dart
class RadiologyResult {
  const RadiologyResult({
    required this.id,
    required this.orderId,
    required this.resultType,
    this.titleAr,
    this.titleEn,
    this.notesAr,
    this.notesEn,
    this.isVisibleToPatient = false,
    this.fileName,
    this.mimeType,
    this.fileSize,
    this.downloadUrl,
    this.uploadedAt,
  });

  final int id;
  final int orderId;
  final String resultType;
  final String? titleAr;
  final String? titleEn;
  final String? notesAr;
  final String? notesEn;
  final bool isVisibleToPatient;
  final String? fileName;
  final String? mimeType;
  final int? fileSize;
  final String? downloadUrl;
  final DateTime? uploadedAt;

  String title(bool isArabic) {
    if (!isArabic && titleEn?.trim().isNotEmpty == true) return titleEn!.trim();
    if (titleAr?.trim().isNotEmpty == true) return titleAr!.trim();
    if (fileName?.trim().isNotEmpty == true) return fileName!.trim();
    return isArabic ? 'نتيجة الأشعة' : 'Radiology result';
  }

  String? notes(bool isArabic) {
    if (!isArabic && notesEn?.trim().isNotEmpty == true) {
      return notesEn!.trim();
    }
    return notesAr;
  }
}

class RadiologyResultDownload {
  const RadiologyResultDownload({
    required this.resultId,
    required this.localPath,
  });

  final int resultId;
  final String localPath;
}

File: etamen-backend/app/Modules/Radiology/Infrastructure/Models/RadiologyResult.php
<?php

namespace App\Modules\Radiology\Infrastructure\Models;

use App\Models\User;
use App\Modules\MedicalFiles\Infrastructure\Models\UploadedFile;
use App\Modules\Radiology\Domain\Enums\RadiologyResultType;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

class RadiologyResult extends Model
{
    protected $fillable = [
        'radiology_order_id',
        'uploaded_file_id',
        'uploaded_by',
        'result_type',
        'title_ar',
        'title_en',
        'notes_ar',
        'notes_en',
        'is_visible_to_patient',
        'uploaded_at',
    ];

    protected function casts(): array
    {
        return [
            'result_type' => RadiologyResultType::class,
            'is_visible_to_patient' => 'boolean',
            'uploaded_at' => 'datetime',
        ];
    }

    public function order(): BelongsTo
    {
        return $this->belongsTo(RadiologyOrder::class, 'radiology_order_id');
    }

    public function file(): BelongsTo
    {
        return $this->belongsTo(UploadedFile::class, 'uploaded_file_id');
    }

    public function uploader(): BelongsTo
    {
        return $this->belongsTo(User::class, 'uploaded_by');
    }
}
